storage/queue.go

```diff
@@ -100,7 +100,7 @@ type baseQueue struct {
 	name       string
 	impl       queueImpl
 	maxSize    int                  // Maximum number of ranges to queue
-	incoming   chan *Range          // Channel for ranges to be queued
+	incoming   chan bool            // Channel signalled when a new range is added to the queue.
```

Comment:
We generally use `chan struct{}` (and `case bq.incoming <- struct{}{}:`) to indicate that the channel just triggers an event and doesn't convey a value (i.e. it doesn't matter whether you send `true` or `false`)